Sources/App/QRCodeScanner/BarcodeScannerView.swift

```diff
@@ -0,0 +1,146 @@
+import CodeScanner
+import Shared
+import SwiftUI
+
+enum QRScannerResult {
+    case cancelled
+    case alternativeOption
+    case success(_ code: String, _ format: String)
+}
+
+struct BarcodeScannerView: View {
+    @Environment(\.dismiss) private var dismiss
+    // Use single data model so both camera previews use same camera stream
+    @State private var cameraDataModel = BarcodeScannerDataModel()
+    private let cameraSquareSize: CGFloat = 320
+    private let flashlightIcon = MaterialDesignIcons.flashlightIcon.image(
+        ofSize: .init(width: 24, height: 24),
+        color: .white
+    )
+
+    private let title: String
+    private let description: String
+    private let alternativeOptionLabel: String?
+    private let completion: (QRScannerResult) -> Void
+
+    init(
+        title: String,
+        description: String,
+        alternativeOptionLabel: String? = nil,
+        completion: @escaping (QRScannerResult) -> Void
+    ) {
+        self.title = title
+        self.description = description
+        self.alternativeOptionLabel = alternativeOptionLabel
+        self.completion = completion
+    }
+
+    var body: some View {
+        ZStack(alignment: .top) {
+            ZStack {
+                cameraBackground
+                cameraSquare
+            }
+            .ignoresSafeArea()
+            .frame(maxWidth: .infinity)
+            .frame(maxHeight: .infinity)
+
+            topInformation
+        }
+        .onAppear {
+            AppDelegate.orientationLock = UIInterfaceOrientationMask.portrait
+            UINavigationController.attemptRotationToDeviceOrientation()
+
+            cameraDataModel.camera.qrFound = { code, format in
+                self.completion(.success(code, format))
+                self.dismiss()
+            }
+        }
+        .onDisappear {
+            DispatchQueue.main.async {
+                AppDelegate.orientationLock = UIInterfaceOrientationMask.all
+                UINavigationController.attemptRotationToDeviceOrientation()
+            }
+        }
+    }
+
+    private var topInformation: some View {
+        VStack(spacing: 8) {
+            Button(action: {
```

Comment:
feelsl ike this needs an accessibility title

Reply:
Good catch